server/odc-service/src/main/java/com/oceanbase/odc/service/permission/database/DatabasePermissionService.java

```diff
@@ -42,32 +77,165 @@
 @Authenticated
 public class DatabasePermissionService {
 
+    @Autowired
+    private ProjectService projectService;
+
+    @Autowired
+    private DatabaseService databaseService;
+
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+
+    @Autowired
+    private PermissionRepository permissionRepository;
+
+    @Autowired
+    private UserPermissionRepository userPermissionRepository;
+
+    @Autowired
+    private UserDatabasePermissionRepository userDatabasePermissionRepository;
+
+    @Autowired
+    private PermissionService permissionService;
+
+    @Value("${odc.iam.permission.expired-retention-time-seconds:7776000}")
+    private long expiredRetentionTimeSeconds;
+
+    private static final UserDatabasePermissionMapper mapper = UserDatabasePermissionMapper.INSTANCE;
+    private static final int EXPIRING_DAYS = 7;
+
     @Transactional(rollbackFor = Exception.class)
-    @PreAuthenticate(hasAnyResourceRole = {"OWNER", "DBA", "DEVELOPER", "SECURITY_ADMINISTRATOR", "PARTICIPANT"},
-            resourceType = "ODC_PROJECT", indexOfIdParam = 0)
+    @SkipAuthorize("permission check inside")
     public Page<UserDatabasePermission> list(@NotNull Long projectId, @NotNull QueryDatabasePermissionParams params,
             Pageable pageable) {
-        throw new NotImplementedException();
+        boolean hasPermission;
+        if (params.getUserId() == null || params.getUserId() != authenticationFacade.currentUserId()) {
```

Comment:
seems we can extract a ProjectPermissionValidator component, for simpiliy related permission verify